rosetta-client/src/tx_builder.rs

```diff
@@ -26,31 +27,28 @@ impl GenericTransactionBuilder {
     pub fn transfer(&self, address: &Address, amount: u128) -> Result<GenericMetadataParams> {
         Ok(match self {
             Self::Astar(tx) => AstarMetadataParams(tx.transfer(address, amount)?).into(),
-            Self::Ethereum(tx) => tx.transfer(address, amount)?.into(),
+            // Self::Ethereum(tx) => tx.transfer(address, amount)?.into(),
             Self::Polkadot(tx) => tx.transfer(address, amount)?.into(),
         })
     }
 
     pub fn method_call(
         &self,
-        contract: &str,
-        method: &str,
-        params: &[String],
+        contract: &[u8; 20],
+        data: &[u8],
         amount: u128,
     ) -> Result<GenericMetadataParams> {
         Ok(match self {
-            Self::Astar(tx) => {
-                AstarMetadataParams(tx.method_call(contract, method, params, amount)?).into()
-            },
-            Self::Ethereum(tx) => tx.method_call(contract, method, params, amount)?.into(),
-            Self::Polkadot(tx) => tx.method_call(contract, method, params, amount)?.into(),
+            Self::Astar(tx) => AstarMetadataParams(tx.method_call(contract, data, amount)?).into(),
+            // Self::Ethereum(tx) => tx.method_call(contract, data, amount)?.into(),
```

Comment:
why is ethereum commented out?

Reply:
Fixed, I was checking if I could add support to Polygon/Arbitrum without adding new items to the enum.